extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind Blob Storage type parameters.
+    /// </summary>
+    internal class BlobStorageConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.Source is CollectionModelBindingData collectionBindingData)
+            {
+                if (collectionBindingData is not null)
+                {
+                    var collectionResult = ConvertCollectionModelBindingDataAsync(context, context.TargetType, collectionBindingData);
+
+                    if (collectionResult is not null && collectionResult.Any())
+                    {
+                        return new ValueTask<ConversionResult>(ConversionResult.Success(collectionResult));
+                    }
+                }
+            }
+            else if (context.Source is ModelBindingData bindingData)
+            {
+                var result = ConvertModelBindingDataAsync(context, context.TargetType, bindingData);
+
+                if (result is not null)
+                {
+                    return new ValueTask<ConversionResult>(ConversionResult.Success(result));
+                }
+            }
+
+            return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+        }
+
+
+        private object? ConvertModelBindingDataAsync(ConverterContext context, Type targetType, ModelBindingData bindingData)
+        {
+            if (bindingData.Source is not Constants.BlobExtensionName)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            if (!TryGetBindingDataContent(bindingData, out IDictionary<string, string> content))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            content.TryGetValue(Constants.Connection, out var connectionName);
+            content.TryGetValue(Constants.ContainerName, out var containerName);
+            content.TryGetValue(Constants.BlobName, out var blobName);
+
+            if (string.IsNullOrEmpty(connectionName) || string.IsNullOrEmpty(containerName))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var connectionString = connectionName is null ? null : Environment.GetEnvironmentVariable(connectionName);
```

Comment:
Is `connectionName is null` necessary?  Is line 67 already not handing that case?